plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -415,6 +422,211 @@ createWidget("discourse-poll-info", {
+      bootbox.alert(I18n.t("poll.error_while_fetching_voters"));
+    }
+  });
+}
+
+function transformUserFieldToLabel(fieldName) {
+  let transformed = fieldName.split("_");
+  transformed[0] =
+    transformed[0].charAt(0).toUpperCase() + transformed[0].slice(1);
+  return transformed.join(" ");
+}
+
+createWidget("discourse-poll-grouped-pies", {
+  tagName: "div.poll-grouped-pies",
+  buildAttributes(attrs) {
+    return {
+      id: `poll-results-grouped-pie-charts-${attrs.id}`
+    };
+  },
+
+  html(attrs) {
+    const fields = Object.assign({}, attrs.groupableUserFields);
+    later(() => {
+      attrs.groupedBy = attrs.groupedBy || fields[0];
+      const parent = document.getElementById(
+        `poll-results-grouped-pie-charts-${attrs.id}`
+      );
+      const fieldSelectId = `field-select-${attrs.id}`;
+
+      const existingSel = document.getElementById(fieldSelectId);
+      if (existingSel) {
+        existingSel.parentNode.removeChild(existingSel);
+      }
+
+      const groupBySelect = document.createElement("select");
+      groupBySelect.classList.add("poll-group-by-selector");
+      groupBySelect.id = fieldSelectId;
+
+      const clearFix = document.createElement("div");
+      clearFix.classList.add("clear-both");
+
+      parent.prepend(groupBySelect);
+      parent.append(clearFix);
+
+      for (var i = 0; i < attrs.groupableUserFields.length; i++) {
+        const opt = document.createElement("option");
+        opt.innerHTML = transformUserFieldToLabel(fields[i]);
+        opt.value = fields[i];
+        groupBySelect.appendChild(opt);
+      }
+      groupBySelect.value = attrs.groupedBy;
+      fetchGroupedResults({
+        post_id: attrs.post.id,
+        poll_name: attrs.poll.name,
+        user_field_name: attrs.groupedBy
+      }).then(result => {
+        for (
+          let chartIdx = 0;
+          chartIdx < result.grouped_results.length;
+          chartIdx++
+        ) {
+          const data = result.grouped_results[chartIdx].options.map(
+            o => o.votes
+          );
+          const labels = result.grouped_results[chartIdx].options.map(
+            o => o.html
+          );
+          const chartConfig = pieChartConfig(data, labels, 1.2);
+          const canvasId = `pie-${attrs.id}-${chartIdx}`;
+          let el = document.querySelector(`#${canvasId}`);
+          if (!el) {
+            const container = document.createElement("div");
+            container.classList.add("poll-grouped-pie-container");
+
+            const label = document.createElement("label");
+            label.classList.add("poll-pie-label");
+            label.textContent = result.grouped_results[chartIdx].group;
+
+            const canvas = document.createElement("canvas");
+            canvas.classList.add(`poll-grouped-pie-${attrs.id}`);
+            canvas.id = canvasId;
+
+            container.appendChild(label);
+            container.appendChild(canvas);
+            parent.appendChild(container);
```

Comment:
I wonder if it would make more sense to have the widget emit the HTML and then only perform the `Chart` in `later()`? Did you try that approach?